include/flatbuffers/flatbuffers.h

```diff
@@ -960,9 +960,11 @@ FLATBUFFERS_FINAL_CLASS
   }
 
   /// @brief Store a string in the buffer, which can contain any binary data.
-  /// @param[in] str A const reference to a std::string to store in the buffer.
+  /// @param[in] str A const reference to a std::string like type with support
+  /// of T::c_str and T::length() to store in the buffer.
   /// @return Returns the offset in the buffer where the string starts.
-  Offset<String> CreateString(const std::string &str) {
+  template<typename T>
+  Offset<String> CreateString(const T &str) {
```

Comment:
don't we want this tho? i.e.: a type that has T::c_str() and T::length() which includes:

* basic_string<character_type=char> i.e.: https://gcc.gnu.org/onlinedocs/gcc-6.2.0/libstdc++/api/a01076_source.html

and most other string types that simplify this interface?

How do you propose we bound the user supplied string so that we can convert it into a flatbuffers::String